src/App.tsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import {
+  getAll,
+  get5First,
+  getRedGoods,
+} from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadGoodsFromServer = async (
+    promiseFromServer: () => Promise<Good[]>,
```

Comment:
will be better to create more understandable variable name: it's not promise from server, it's function that should create corresponding request to the server

Reply:
you are right